ProcessLib/BoundaryCondition/HCNonAdvectiveFreeComponentFlowBoundaryConditionLocalAssembler.h

```diff
@@ -0,0 +1,118 @@
+/**
+ * \copyright
+ * Copyright (c) 2012-2019, OpenGeoSys Community (http://www.opengeosys.org)
+ *            Distributed under a Modified BSD License.
+ *              See accompanying file LICENSE.txt or
+ *              http://www.opengeosys.org/project/license
+ *
+ */
+
+#pragma once
+
+#include "GenericNaturalBoundaryConditionLocalAssembler.h"
+#include "MeshLib/Elements/MapBulkElementPoint.h"
+#include "MeshLib/PropertyVector.h"
+#include "NumLib/DOF/DOFTableUtil.h"
+#include "NumLib/Function/Interpolation.h"
+#include "ParameterLib/MeshNodeParameter.h"
+#include "ProcessLib/Process.h"
+
+namespace ProcessLib
+{
+struct HCNonAdvectiveFreeComponentFlowBoundaryConditionData
+{
+    ParameterLib::Parameter<double> const& boundary_permeability;
+    MeshLib::PropertyVector<std::size_t> const bulk_face_ids;
+    MeshLib::PropertyVector<std::size_t> const bulk_element_ids;
```

Comment:
Not sure at this point (of review) but is copying the property vectors intended? I'd guess a const-ref is sufficient.